src/java/main/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -486,19 +474,27 @@ protected void startSessionTracker() {
         ((SessionTrackerImpl)sessionTracker).start();
     }
 
+    void setState(State state) {
```

Comment:
I'd really appreciate if we could have a fair amount of comments for these new methods explaining why they are here and how they are used. The methods themselves are trivial, but the whole logic of stoping servers due to errors and such isn't necessarily trivial.